### PR TITLE
[coq] Adapt to coq/coq#19187

### PR DESCRIPTION
**Pull request by ejgallego**

### CODE REVIEW

**Review by gares — CHANGES_REQUESTED** [review]
Please use outcome, that is put the change in a [%%if...] block.

File: language-server/dm/documentManager.ml [hunk 1]
```diff
@@ -421,8 +421,8 @@ let get_completions st pos =
 let parse_entry st pos entry pattern =
   let pa = Pcoq.Parsable.make (Gramlib.Stream.of_string pattern) in
   let st = match Document.find_sentence_before st.document pos with
-  | None -> st.init_vs.Vernacstate.synterp.parsing
-  | Some { synterp_state } -> synterp_state.Vernacstate.Synterp.parsing
+  | None -> Vernacstate.(Synterp.parsing st.init_vs.synterp)
+  | Some { synterp_state } -> Vernacstate.Synterp.parsing synterp_state
```
Comment (SkySkimmer) [inline language-server/dm/documentManager.ml:425]:
```suggestion
[%%if coq = "8.20"]
  | None -> Vernacstate.(Synterp.parsing st.init_vs.synterp)
  | Some { synterp_state } -> Vernacstate.Synterp.parsing synterp_state
[%%else]
  | None -> st.init_vs.Vernacstate.synterp.parsing
  | Some { synterp_state } -> synterp_state.Vernacstate.Synterp.parsing
[%%endif]
```

File: language-server/dm/documentManager.ml [hunk 2]
```diff
@@ -421,8 +421,13 @@ let get_completions st pos =
...
   let st = match Document.find_sentence_before st.document pos with
+[%%if coq = "8.20"]
+  | None -> Vernacstate.(Synterp.parsing st.init_vs.synterp)
+  | Some { synterp_state } -> Vernacstate.Synterp.parsing synterp_state
+[%%else]
   | None -> st.init_vs.Vernacstate.synterp.parsing
   | Some { synterp_state } -> synterp_state.Vernacstate.Synterp.parsing
+[%%endif]
   in
   Vernacstate.Parser.parse st entry pa
```
Comment (SkySkimmer) [inline language-server/dm/documentManager.ml:432]:
```suggestion
[%%if coq = "8.20"]
let parse_entry st pos entry pattern =
  let pa = Pcoq.Parsable.make (Gramlib.Stream.of_string pattern) in
  let st = match Document.find_sentence_before st.document pos with
  | None -> Vernacstate.(Synterp.parsing st.init_vs.synterp)
  | Some { synterp_state } -> Vernacstate.Synterp.parsing synterp_state
  in
  Vernacstate.Parser.parse st entry pa
[%%else]
let parse_entry st pos entry pattern =
  let pa = Pcoq.Parsable.make (Gramlib.Stream.of_string pattern) in
  let st = match Document.find_sentence_before st.document pos with
  | None -> st.init_vs.Vernacstate.synterp.parsing
  | Some { synterp_state } -> synterp_state.Vernacstate.Synterp.parsing
  in
  Vernacstate.Parser.parse st entry pa
[%%endif]
```

**Comment by ejgallego** [comment]
Thanks @SkySkimmer

**Pull request merged**